src/Components/Blueprints/BlueprintsSideBar.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, Dispatch, SetStateAction } from 'react';
+
+import {
+  EmptyState,
+  EmptyStateActions,
+  EmptyStateBody,
+  EmptyStateFooter,
+  EmptyStateHeader,
+  EmptyStateIcon,
+  SearchInput,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Link } from 'react-router-dom';
+
+import BlueprintCard from './BlueprintCard';
+
+import { BlueprintItem } from '../../store/imageBuilderApi';
+import { resolveRelPath } from '../../Utilities/path';
+
+interface blueprintProps {
+  blueprints: BlueprintItem[] | undefined;
+  selectedBlueprint: string;
+  setSelectedBlueprint: Dispatch<SetStateAction<string>>;
+}
+const BlueprintsSidebar = (props: blueprintProps) => {
+  const [blueprintFilter, setBlueprintFilter] = useState('');
+
+  const onChange = (value: string) => {
+    setBlueprintFilter(value);
+  };
+
+  const blueprintsEmpty = (
+    <EmptyState variant="sm">
+      <EmptyStateHeader
+        titleText="No blueprints yet"
+        headingLevel="h4"
+        icon={<EmptyStateIcon icon={PlusCircleIcon} />}
+      />
+      <EmptyStateBody>To get started, create a blueprint.</EmptyStateBody>
+      <EmptyStateFooter>
+        <EmptyStateActions>
+          <Link
+            to={resolveRelPath('blueprintwizard')}
+            className="pf-c-button pf-m-primary pf-u-mr-md"
+            data-testid="create-blueprint-action"
+          >
+            Create
+          </Link>
+        </EmptyStateActions>
+      </EmptyStateFooter>
+    </EmptyState>
+  );
+
+  return props.blueprints?.length ? (
```

Comment:
To be more consistent with the rest of the code base, let's use short circuits here for conditional rendering, e.g.
```typescript
return (
  {blueprints?.length > 0 && (<> ... </>)}
  {blueprints === undefined || blueprints.length === 0 && (<EmptyState>)
)
```

Reply:
And fwiw both are acceptable and shown in the official React docs, we just tend to use the short circuit method:
https://react.dev/learn/conditional-rendering#logical-and-operator-

Reply:
this example with the && operator in the same return segment,  ends up with unreachable code.

Reply:
Cool, yeah that was just a quick off the cuff example not supposed to be final code :sweat_smile: thanks for fixing it up.